Objects/TI4_Helpers/TI4_FactionHelper.ttslua

```diff
@@ -1122,6 +1126,703 @@ function tintTokens(params)
+
+    --Find all units on those hexes
+    local opponentShipToPosition = {}
+    local unitGuidToUnit = {}
+    local allUnits = _unitHelper.getUnits()
+    for _, unit in ipairs(allUnits) do
+        if unit.unitType ~= 'Infantry' and unit.unitType ~= 'Mech' and unit.unitType ~= 'Fighter' and unit.unitType ~= 'Space Dock' and unit.unitType ~= 'PDS' then --non-fighter ships only
+            if (unit.factionTokenName and unit.factionTokenName ~= factionTokenName) or (unit.color and unit.color ~= factionColor) then
+                opponentShipToPosition[unit.guid] = unit.position
+                unitGuidToUnit[unit.guid] = unit
+            end
+        end
+    end
+
+    local opponentShipGuidToHex = _systemHelper.hexesFromPositions(opponentShipToPosition)
+    local opponentShipToHex = {}
+    for guid, hex in pairs(opponentShipGuidToHex) do
+        opponentShipToHex[unitGuidToUnit[guid]] = opponentShipGuidToHex[guid]
+    end
+
+    local anyAnchoredShips = false
+    local systemToAnchoredShips = {}
+    for unit, hex in pairs(opponentShipToHex) do
+        local system = allHexesToSystemTiles[hex]
+        if system then
+            local systemShips = systemToAnchoredShips[system]
+            if not systemShips then
+                systemShips = {}
+                systemToAnchoredShips[system] = systemShips
+            end
+
+            table.insert(systemShips, unit)
+            anyAnchoredShips = true
+        end
+    end
+
+    if not anyAnchoredShips then
+        print('Dimensional Anchor: No non-fighter ships adjacent to Dimensional Tear systems. Won\'t use Hero Ability.')
+        return 1
+    end
+
+    -- Safe to commit to acting at this point, so purge the Hero card.
+    if owningObject.tag == 'Card' then
+        purgeCard(owningObject)
+    end
+
+    -- Function spawns dice for each ship, sets a dice cleanup on a 2 minute timeout,
+    -- and prints a guide for the dice color for auditability.
+    local function _prepareDice(systemToAnchoredShips)
+        -- Data and methods for dice handling copied from multiroller.
+
+        local dieType = "Die_10"
+        local removalDelay = 5
+        local radialOffset = 1.2
+        local heightOffset = 5
+        local dieSize = 1
+
+        --Finds a position, rotated around the Y axis, using distance you want + angle
+        --oPos is object pos, oRot=object rotation, distance = how far, angle = angle in degrees
+        local function _findGlobalPosWithLocalDirection(spawn_object, angle)
+            local object, distance = spawn_object, radialOffset * self.getScale().x
+            local oPos, oRot = object.getPosition(), object.getRotation()
+            local posX = oPos.x + math.sin( math.rad(angle+oRot.y) ) * distance
+            local posY = oPos.y + heightOffset
+            local posZ = oPos.z + math.cos( math.rad(angle+oRot.y) ) * distance
+            return {x=posX, y=posY, z=posZ}
+        end
+
```

Comment:
Seems like there are a few places that roll dice (more than one with per-unit die coloring).  More of a note to self but it might be worth making a new helper (with some kind of callback?).